.github/workflows/codeql-analysis.yml

```diff
@@ -27,11 +27,8 @@ name: "CodeQL"
 on:
   push:
     branches: [ main ]
-  pull_request:
```

Comment:
Why is the PR check removed? We will not catch if there are any issues in the PR.

Reply:
Since codeQL realistically never finds any results and makes pre-submit slower, we've been thinking of running in only on daily schedules, and use an API like  https://docs.github.com/en/rest/code-scanning?apiVersion=2022-11-28 to retrieve result and alert us (create an issue or as pre-submit) if codeQL has found anything.

Reply:
OK, sounds good. Thanks